python/cudf/cudf/core/dataframe.py

```diff
@@ -2172,8 +2180,10 @@ def transpose(self):
         Difference from pandas:
         Not supporting *copy* because default and only behaviour is copy=True
         """
-        # Never transpose a MultiIndex - remove the existing columns and
-        # replace with a RangeIndex. Afterward, reassign.
+        if isinstance(self.index, cudf.MultiIndex):
+            warnings.warn("Never transpose a MultiIndex - remove the "
+                "existing columns and replace with a "
+                "RangeIndex. Afterward, reassign.", Warning)
```

Comment:
So is it fine that I removed the groupby repr transform test? We don't have matching functionality with pandas in this case. I think I can resolve that though, I was thinking this was more strict but it appears that we break this rule everywhere.

Reply:
I am unsure of how to approach this issue with the `DataFrameT.__repr__()` if we don't transpose `MultiIndex` without adding logic to `DataFrame.transpose()` that I feel would be undesirable. I commented it out, but I am open to other suggestions.

```
@pytest.mark.parametrize("nrows", [0, 1, 3, 5, 10])
@pytest.mark.parametrize("ncols", [0, 1, 2, 3])
def test_groupby_MI(nrows, ncols):
    gdf = cudf.DataFrame(
        {"a": np.arange(10), "b": np.arange(10), "c": np.arange(10)}
    )
    pdf = gdf.to_pandas()
    gdg = gdf.groupby(["a", "b"]).count()
    pdg = pdf.groupby(["a", "b"]).count()
    pd.options.display.max_rows = nrows
    pd.options.display.max_columns = ncols
    assert gdg.__repr__() == pdg.__repr__()
    # assert gdg.T.__repr__() == pdg.T.__repr__()
```

Reply:
What's the issue being run into? Could you explain why we can't handle transposing a MultiIndex currently?

Reply:
I can do it in the test or I can handle it in `Dataframe.transpose`. Ideally it would be handled in `transpose()` I think. Maybe I'm being dense, but we have multiple places already doing the `remove the existing columns and replace with a RangeIndex. Afterward, reassign.` Logically I would think we should be detecting in `transpose()` but I was maybe erroneously thinking that was not what we wanted. I don't know how I got that idea, I think I was just gun shy to add checks to `transpose()` because there's a comment and we haven't. Is there any reason not to just detect and do the work in `transpose()` and remove any duplicate logic in other places?